pandas/tests/series/methods/test_drop_duplicates.py

```diff
@@ -224,6 +225,20 @@ def test_drop_duplicates_categorical_bool(self, ordered):
         assert return_value is None
         tm.assert_series_equal(sc, tc[~expected])
 
+    def test_drop_duplicates_categorical_bool_na(self):
+        # GH#44351
+        ser = Series(
+            Categorical(
+                [True, False, True, False, NA], categories=[True, False], ordered=True
+            )
+        )
+        result = ser.drop_duplicates()
+        expected = Series(
+            Categorical([True, False, np.nan], categories=[True, False], ordered=True),
```

Comment:
this might be a bug as this is not preserving NA (but unrelated / not a regression, so pls open a new issue)

Reply:
Done, #44405

Reply:
The test case is actually a bit misleading. This regression was just about a boolean categorical with missing values, not specifically with `NA` (also if you create the data with np.nan, you had the same issue)